### PR TITLE
generate socket.ini documentation

### PR DESCRIPTION
**Pull request by chicoxyzzy**
Undocumented stuff from config:
- [x] `"android_" + platform.os + "_home"`
- [x] `android_home`
- [x] `android_enable_standard_ndk_build`
- [x] `android_skip_gradle`
- [x] `android_build_socket_runtime`
- [x] `android_main_activity`
- [x] `android_sdk_manager_path`
- [x] `android_aapt_no_compress`
- [x] `android_aapt` (undocumented)
- [x] `android_native_abis`
- [x] `android_ndk_abi_filters` (undocumented)
- [x] `android_manifest_permissions`
- [x] `android_allow_cleartext`
- [x] `android_default_config_external_native_build`
- [x] `android_external_native_build`
- [x] `android_native_cflags`
- [x] `android_native_sources`
- [x] `android_native_makefile`
- [x] `android_sources`
- [x] `android_sdk_manager_path`
- [x] `ios_provisioning_specifier`
- [x] `ios_provisioning_profile`
- [x] `ios_codesign_identity`
- [x] `apple_team_id`
- [x] `apple_instruments`
- [x] `headless_runner`
- [x] `headless_runner_flags`
- [x] `"headless_" + platform.os + "_runner"`
- [x] `"headless_" + platform.os + "_runner_flags"`
- [x] `build_suffix`
- [x] `linux_executable_path`
- [x] `deb_name`
- [x] `win_version`
- [x] `exe`
- [x] `mac_sign`
- [x] `mac_sign_paths`

### CODE REVIEW

File: src/cli/cli.cc [hunk 1]
```diff
@@ -1277,7 +1269,6 @@ int main (const int argc, const char* argv[]) {
           }
 
           settings["debug"] = flagDebugMode;
-          settings["build_suffix"] = suffix;
```
Comment (chicoxyzzy) [inline src/cli/cli.cc:1280]: not used anywhere

File: src/cli/cli.cc [hunk 2]
```diff
@@ -165,33 +165,19 @@ String getSocketHome () {
...
-  if (androidHome.size() > 0)
+  if (androidHome.size() > 0) {
     return androidHome;
-
-  if (androidHome.size() == 0) {
-    androidHome = settings["android_" + platform.os + "_home"];
   }
 
-  if (androidHome.size() == 0) {
-    androidHome = settings["android_home"];
```
Comment (chicoxyzzy) [inline src/cli/cli.cc:181]: removed in favor of `.ssc.env` and `.sscrc`

File: src/cli/cli.cc [hunk 3]
```diff
@@ -2445,10 +2437,6 @@ int main (const int argc, const char* argv[]) {
 
       writeFile(pathManifestFile / (settings["build_name"] + ".desktop"), tmpl(gDestkopManifest, settings));
 
-      if (settings.count("deb_name") == 0) {
-        settings["deb_name"] = settings["build_name"];
-      }
-
```
Comment (chicoxyzzy) [inline src/cli/cli.cc:2451]: just use `build_name` in the deb template

**Comment by mribbons** [comment]
@chicoxyzzy Can you rebase to `master` and see if Android / Linux CI works now that https://github.com/socketsupply/socket/pull/360 has been merged?

**Comment by chicoxyzzy** [comment]
@jwerle @mribbons still need your help with last 4 🙏

**Review by heapwolf — APPROVED** [review]

**Pull request merged**